includes/Core/Modules/Module.php

```diff
@@ -518,15 +518,29 @@ private function validate_data_request( Data_Request $data ) {
 			return;
 		}
 
-		$datapoint = $definitions[ $datapoint_key ];
-
-		// If the datapoint requires specific scopes, ensure they are satisfied.
-		if ( ! $this->authentication->get_oauth_client()->has_sufficient_scopes( $datapoint['scopes'] ) ) {
-			$request_scopes_message = ! empty( $datapoint['request_scopes_message'] )
-				? $datapoint['request_scopes_message']
-				: __( 'You’ll need to grant Site Kit permission to do this.', 'google-site-kit' );
+		$datapoint        = $definitions[ $datapoint_key ];
+		$requires_service = isset( $datapoint['service'] );
+		$scopes           = $datapoint['scopes'];
```

Comment:
`$datapoint[scopes]` are datapoint-specific scopes, which are the on-demand write scopes (e.g. to create a property/profile, etc).

This issue is about making sure that the module's "base scopes" (i.e. non-datapoint-specific read-only scopes) are fulfilled if the datapoint depends on a service. _These scopes_ are what is returned by `Module::get_scopes` (in this context `$this->get_scopes()`).

So the existing datapoint-specific scope check should be done first, then check the base scopes.